### ISSUE TITLE
TypeScript errors

### ISSUE THREAD

**Issue by benc-uk** [prantlf/jsonlint#7]
I discovered how old the JSONLint module I was using, so thought I'd try your fork out
My project is TypeScript based and I'm seeing these weird errors

```
node_modules/@prantlf/jsonlint/lib/jsonlint.d.ts:57:16 - error TS2665: Invalid module name in augmentation. Module '@prantlf/jsonlint/lib/validator' resolves to an untyped module at '/home/ben/dev/armview-vscode/node_modules/@prantlf/jsonlint/lib/validator.js', which cannot be augmented.

57 declare module '@prantlf/jsonlint/lib/validator' {
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@prantlf/jsonlint/lib/jsonlint.d.ts:88:16 - error TS2665: Invalid module name in augmentation. Module '@prantlf/jsonlint/lib/printer' resolves to an untyped module at '/home/ben/dev/armview-vscode/node_modules/@prantlf/jsonlint/lib/printer.js', which cannot be augmented.

88 declare module '@prantlf/jsonlint/lib/printer' {
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by benc-uk** [comment]
I don't think you can have an "@" in a module name

**Comment by benc-uk** [comment]
Also this https://github.com/Microsoft/TypeScript/issues/16785#issuecomment-323223478

**Comment by prantlf** [comment]
Thanks for reporting this! I'm sorry for the late answer. I fixed the problem in [11.7.2](https://github.com/prantlf/jsonlint/releases/tag/v11.7.2).

* Complete TypeScript types ([7064c50](https://github.com/prantlf/jsonlint/commit/7064c5041a292a5a87bccc2de7fc945a2ee7c160))
* Use both typings and types in package.json ([5d00c00](https://github.com/prantlf/jsonlint/commit/5d00c00c7fd098674ee9d1f3dba14369debaa73b))

I merged all types to the single file `index.d.ts` and declared only modules in its root, as one of the comments in the issue which you referred to suggested.

The compilation of a dummy project succeeds and the types are recognised in VS Code properly.

package.json:

```json
{
  "private": true,
  "devDependencies": {
    "@prantlf/jsonlint": "11.7.2",
    "typescript": "4.9.5"
  }
}
```

index.ts:

```ts
import { parse } from '@prantlf/jsonlint'
import { compile } from '@prantlf/jsonlint/lib/validator'
import { print } from '@prantlf/jsonlint/lib/printer'

parse('', { ignoreBOM: true })
compile('', { environment: 'json-schema-draft-04' })
print([], { indent: '  ' })
```

commands:

    pnpm i
    npx tsc --lib es6 index.ts
